homeassistant/components/lyric/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+from aiolyric.objects.location import LyricLocation
+from aiolyric.objects.priority import LyricRoom
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import LyricRoomEntity
+from .const import DOMAIN
+
+
+@dataclass(frozen=True)
+class LyricRoomBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[LyricRoom], bool]
+
+
+@dataclass(frozen=True)
+class LyricRoomBinarySensorEntityDescription(
+    BinarySensorEntityDescription, LyricRoomBinarySensorEntityDescriptionMixin
+):
+    """Class describing Honeywell Lyric room binary sensor entities."""
+
+
+ROOM_SENSORS = [
+    LyricRoomBinarySensorEntityDescription(
+        key="overall_motion",
+        translation_key="overall_motion",
+        name="overall motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value_fn=lambda room: room.overallMotion,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Honeywell Lyric binary sensor platform based on a config entry."""
+    coordinator: DataUpdateCoordinator[Lyric] = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+
+    for location in coordinator.data.locations:
+        for device in location.devices:
+            for room in coordinator.data.rooms_dict[device.macID].values():
+                for room_sensor in ROOM_SENSORS:
+                    entities.append(
+                        LyricRoomBinarySensor(
+                            coordinator,
+                            room_sensor,
+                            location,
+                            device,
+                            room,
+                        )
+                    )
+
+    async_add_entities(entities)
+
+
+class LyricRoomBinarySensor(LyricRoomEntity, BinarySensorEntity):
+    """Define a Honeywell Lyric room binary sensor."""
+
+    entity_description: LyricRoomBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Lyric],
+        description: LyricRoomBinarySensorEntityDescription,
+        location: LyricLocation,
+        device: LyricDevice,
+        room: LyricRoom,
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            room,
+            f"{device.macID}_room_{room.id}_{description.key}",
+        )
+        self.entity_description = description
+        room_name = room.roomName or f"Room {room.id}"
+        self._attr_name = f"{room_name} {description.name}"
```

Comment:
Why is the name composed like this? What's the device's name?

Reply:
Great question: these entities represent the state of individual sensors that are placed in each room. However, the data is not being reported from those sensor devices directly; it's being collected and reported by the thermostat device. So the device might be "Living Room Thermostat", but the entity might represent a sensor in the Kitchen. I don't want these entities showing up as "Living Room Kitchen Temperature" because that doesn't make sense, so I'm omitting the device name and using the room name that is configured in Lyric instead.

If you can think of a better way to do this I'm open to it!